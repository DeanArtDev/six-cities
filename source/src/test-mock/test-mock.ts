const mockOffers = [
  {
    id: 333,
    title: `Beautiful & luxurious apartment at great location`,
    typeHousing: `apartment`,
    price: 200,
    isPremium: false,
    features: [`Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    bedrooms: 2,
    maxAdults: 4,
    isFavorite: true,
    host: {
      avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    city: {
      coordinates: [48.85661, 2.351499],
      zoom: 13,
      name: `Paris`,
    },
    location: {
      coordinates: [52.3909553943508, 4.929309666406198],
      zoom: 16,
    },
    description: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      Aliquam aut distinctio ex recusandae. Debitis fugiat inventore itaque magni mollitia pariatur reprehenderit sapiente ullam.
      Cumque ducimus, est iusto nam natus neque nostrum quae veniam. A culpa eaque in laboriosam odio repellendus sequi sit voluptates.
      Ad adipisci architecto assumenda at autem consectetur consequuntur culpa, cumque debitis earum et eum ipsa iure nam,
      officiis placeat, quae qui quidem ratione rem tempore ut voluptate.`,
    rating: 1.7,
    images: [`apartment-01.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `room.jpg`, `studio-photos.jpg`, `studio-01.jpg`],
    prevImage: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
    isProStatus: false,
  },
  {
    id: 777,
    title: `Beautiful & luxurious apartment at great location`,
    typeHousing: `apartment`,
    price: 200,
    isPremium: false,
    features: [`Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    bedrooms: 2,
    maxAdults: 4,
    isFavorite: true,
    host: {
      avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    city: {
      coordinates: [48.85661, 2.351499],
      zoom: 13,
      name: `Paris`,
    },
    location: {
      coordinates: [52.3909553943508, 4.929309666406198],
      zoom: 16,
    },
    description: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      Aliquam aut distinctio ex recusandae. Debitis fugiat inventore itaque magni mollitia pariatur reprehenderit sapiente ullam.
      Cumque ducimus, est iusto nam natus neque nostrum quae veniam. A culpa eaque in laboriosam odio repellendus sequi sit voluptates.
      Ad adipisci architecto assumenda at autem consectetur consequuntur culpa, cumque debitis earum et eum ipsa iure nam,
      officiis placeat, quae qui quidem ratione rem tempore ut voluptate.`,
    rating: 1.7,
    images: [`apartment-01.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `room.jpg`, `studio-photos.jpg`, `studio-01.jpg`],
    isProStatus: false,
    prevImage: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
  },
];

export const TestMock = {
  offers: [
    {
      city: {
        name: `Brussels`,
        location: {
          latitude: 50.846557,
          longitude: 4.351697,
          zoom: 13
        }
      },
      // eslint-disable-next-line camelcase
      preview_image: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg`,
      images: [`https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg`],
      title: `Waterfront with extraordinary view`,
      // eslint-disable-next-line camelcase
      is_favorite: false,
      // eslint-disable-next-line camelcase
      is_premium: false,
      rating: 3.7,
      type: `hotel`,
      bedrooms: 3,
      // eslint-disable-next-line camelcase
      max_adults: 6,
      price: 109,
      goods: [`Towels`, `Washer`, `Air conditioning`, `Cable TV`, `Washing machine`, `Fridge`, `Breakfast`],
      host: {
        id: 25,
        name: `Angelina`,
        // eslint-disable-next-line camelcase
        is_pro: true,
        // eslint-disable-next-line camelcase
        avatar_url: `img/avatar-angelina.jpg`
      },
      // eslint-disable-next-line camelcase
      id: 33,
      name: `Angelina`,
      // eslint-disable-next-line camelcase
      is_pro: true,
      // eslint-disable-next-line camelcase
      avatar_url: `img/avatar-angelina.jpg`,
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      location: {
        latitude: 50.837557,
        longitude: 4.339697,
        zoom: 16
      },
      latitude: 50.837557,
      longitude: 4.339697,
      zoom: 16,
    },
    {
      city: {
        name: `Paris`,
        location: {
          latitude: 50.846557,
          longitude: 4.351697,
          zoom: 13
        }
      },
      // eslint-disable-next-line camelcase
      preview_image: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg`,
      images: [`https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg`],
      title: `Waterfront with extraordinary view`,
      // eslint-disable-next-line camelcase
      is_favorite: false,
      // eslint-disable-next-line camelcase
      is_premium: false,
      rating: 3.7,
      type: `hotel`,
      bedrooms: 3,
      // eslint-disable-next-line camelcase
      max_adults: 6,
      price: 109,
      goods: [`Towels`, `Washer`, `Air conditioning`, `Cable TV`, `Washing machine`, `Fridge`, `Breakfast`],
      host: {
        id: 25,
        name: `Angelina`,
        // eslint-disable-next-line camelcase
        is_pro: true,
        // eslint-disable-next-line camelcase
        avatar_url: `img/avatar-angelina.jpg`
      },
      // eslint-disable-next-line camelcase
      id: 22,
      name: `Angelina`,
      // eslint-disable-next-line camelcase
      is_pro: true,
      // eslint-disable-next-line camelcase
      avatar_url: `img/avatar-angelina.jpg`,
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      location: {
        latitude: 50.837557,
        longitude: 4.339697,
        zoom: 16
      },
      latitude: 50.837557,
      longitude: 4.339697,
      zoom: 16,
    },
  ],
  adaptedOffers: [
    {
      id: 333,
      title: `Beautiful & luxurious apartment at great location`,
      typeHousing: `apartment`,
      price: 200,
      isPremium: false,
      features: [`Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
      bedrooms: 2,
      maxAdults: 4,
      isFavorite: true,
      host: {
        avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
        id: 25,
        isPro: true,
        name: `Angelina`,
      },
      city: {
        coordinates: [48.85661, 2.351499],
        zoom: 13,
        name: `Paris`,
      },
      location: {
        coordinates: [52.3909553943508, 4.929309666406198],
        zoom: 16,
      },
      description: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      Aliquam aut distinctio ex recusandae. Debitis fugiat inventore itaque magni mollitia pariatur reprehenderit sapiente ullam.
      Cumque ducimus, est iusto nam natus neque nostrum quae veniam. A culpa eaque in laboriosam odio repellendus sequi sit voluptates.
      Ad adipisci architecto assumenda at autem consectetur consequuntur culpa, cumque debitis earum et eum ipsa iure nam,
      officiis placeat, quae qui quidem ratione rem tempore ut voluptate.`,
      rating: 1.7,
      images: [`apartment-01.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `room.jpg`, `studio-photos.jpg`, `studio-01.jpg`],
      prevImage: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
      isProStatus: false,
    },
    {
      id: 777,
      title: `Beautiful & luxurious apartment at great location`,
      typeHousing: `apartment`,
      price: 200,
      isPremium: false,
      features: [`Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
      bedrooms: 2,
      maxAdults: 4,
      isFavorite: true,
      host: {
        avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
        id: 25,
        isPro: true,
        name: `Angelina`,
      },
      city: {
        coordinates: [48.85661, 2.351499],
        zoom: 13,
        name: `Paris`,
      },
      location: {
        coordinates: [52.3909553943508, 4.929309666406198],
        zoom: 16,
      },
      description: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      Aliquam aut distinctio ex recusandae. Debitis fugiat inventore itaque magni mollitia pariatur reprehenderit sapiente ullam.
      Cumque ducimus, est iusto nam natus neque nostrum quae veniam. A culpa eaque in laboriosam odio repellendus sequi sit voluptates.
      Ad adipisci architecto assumenda at autem consectetur consequuntur culpa, cumque debitis earum et eum ipsa iure nam,
      officiis placeat, quae qui quidem ratione rem tempore ut voluptate.`,
      rating: 1.7,
      images: [`apartment-01.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `room.jpg`, `studio-photos.jpg`, `studio-01.jpg`],
      isProStatus: false,
      prevImage: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
    },
  ],
  comments: [
    {
      id: 1,
      user: {
        id: 17,
        // eslint-disable-next-line camelcase
        is_pro: false,
        name: `Emely`,
        // eslint-disable-next-line camelcase
        avatar_url: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/8.jpg`,
      },
      rating: 5,
      comment: `The room was spacious and clean. The pool looked nothing e sauna and spa were closed for lunar new year holiday.`,
      date: `2020-02-12T03:29:40.085Z`,
    }
  ],
  adaptedComments: [
    {
      id: 1,
      rating: 3,
      date: `2020-02-13T01:42:13.129Z`,
      comment: `We loved it so much, the house, the veiw, the location`,
      user: {
        avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg`,
        name: `Mollie`,
        id: 16,
        isPro: false,
      },
    },
    {
      user: {
        id: 13,
        isPro: true,
        name: `Dean`,
        avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg`,
        avatarUrl2: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg`,
      },
      id: 5,
      rating: 1,
      date: `2020-02-13T01:42:13.129Z`,
      comment: `We loved it so much, the house, the veiw, the location`,
    },
  ],
  cities: [`Paris`, `Brussels`, `Hamburg`, `Amsterdam`, `Cologne`],
  coordinates: [
    [52.3909553943508, 4.929309666406198],
    [52.3909553943508, 4.929309666406198],
    [52.3909553943508, 4.929309666406198],
    [52.3909553943508, 4.929309666406198],
  ],
  settingMap: {
    cityCoordinates: [52.3909553943508, 4.929309666406198],
    zoom: 16
  },
  authorizationStatus: {
    AUTH: `AUTH`,
    NOT_AUTH: `NOT_AUTH`,
  },
  authInfo: {
    id: 313,
    email: `email@email.com`,
    name: `You`,
    avatarUrl: `/avatar.svg`,
    isPro: true,
  },
  authInfoNotAdapted: {
    id: 313,
    email: `email@email.com`,
    name: `You`,
    // eslint-disable-next-line camelcase
    avatar_url: `/avatar.svg`,
    // eslint-disable-next-line camelcase
    is_pro: true,
  },
  favoriteOffers: {
    'Paris': mockOffers,
    'Moscow': mockOffers,
  },
  commentPost: {
    "comment": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    "rating": 4
  }
};
