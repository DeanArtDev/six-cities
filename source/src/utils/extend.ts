const extend = (a: object, b: object): object => {
  const emptyObj: object = {};

  return Object.assign(emptyObj, a, b);
};

export default extend;
