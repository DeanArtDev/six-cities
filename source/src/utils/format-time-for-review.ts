import {MONTHS} from '@root/consts/consts';

const formatTimeForReview = (date: string): string => {
  const nextDate = new Date(date);
  const month = MONTHS[nextDate.getMonth()];
  const day = nextDate.getDate() < 10 ? `0${nextDate.getDate()}` : nextDate.getDate();

  return `${month} ${day} ${nextDate.getFullYear()}`;
};

export default formatTimeForReview;
