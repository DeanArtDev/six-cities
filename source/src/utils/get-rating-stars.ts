const ONE_STAR_PRESENTS = 20;

const getRatingStars = (value: number): number | 0 => {
  if (value === 0 || value === undefined || isNaN(value)) {
    return 0;
  }

  return Math.ceil(value) * ONE_STAR_PRESENTS;
};

export default getRatingStars;
