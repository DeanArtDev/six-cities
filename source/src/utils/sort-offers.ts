import {OffersSortTypes} from '@root/types';

const sortOffers = (typeSort, offers) => {
  switch (typeSort) {
    case OffersSortTypes.LOW_TO_HIGH:
      return offers
        .slice()
        .sort((max, min) => max.price - min.price);

    case OffersSortTypes.HIGH_TO_LOW:
      return offers
        .slice()
        .sort((max, min) => min.price - max.price);

    case OffersSortTypes.RATED_FIRST:
      return offers
        .slice()
        .sort((max, min) => min.rating - max.rating);

    default:
      return offers;
  }
};

export default sortOffers;
