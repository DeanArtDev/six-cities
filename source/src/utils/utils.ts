export {default as getRatingStars} from './get-rating-stars';
export {default as extend} from './extend';
export {default as sortOffers} from './sort-offers';
export {default as formatTimeForReview} from './format-time-for-review';
export {default as noop} from './noop';
