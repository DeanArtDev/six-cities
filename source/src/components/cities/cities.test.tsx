import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Cities from './cities';
import {TestMock} from '@root/test-mock/test-mock';
import {Router} from 'react-router-dom';
import history from '@root/history';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {noop} from "@root/utils/utils";

const mockStore = configureStore([]);

describe(`<Cities/>: `, () => {

  const store = mockStore({
    USER: {
      authorizationStatus: TestMock.authorizationStatus.NOT_AUTH,
      authInfo: TestMock.authInfo,
    }
  });

  it(`should render correctly`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router
              history={history}
            >
              <Cities
                offers={TestMock.adaptedOffers}
                currentCity={`Paris`}
                activeCoordinate={TestMock.coordinates[0]}
                onChangerOfferCoordinate={noop}
                onTitleClick={noop}
                coordinates={TestMock.coordinates}
                settingMap={TestMock.settingMap}
                onChangeActive={noop}
                onTypeSort={noop}
                isActive={true}
                typeSort={`test`}
              />
            </Router>
          </Provider>,
          {createNodeMock: () => document.createElement(`section`)}
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
