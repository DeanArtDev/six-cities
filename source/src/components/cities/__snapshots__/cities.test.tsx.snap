// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cities/>:  should render correctly 1`] = `
<div
  className="cities"
>
  <div
    className="cities__places-container container"
  >
    <section
      className="cities__places places"
    >
      <h2
        className="visually-hidden"
      >
        Places
      </h2>
      <b
        className="places__found"
      >
        2
         places to stay in 
        Paris
      </b>
      <form
        action="#"
        className="places__sorting"
        method="get"
      >
        <span
          className="places__sorting-caption"
        >
          Sort by 
        </span>
        <span
          className="places__sorting-type"
          onClick={[Function]}
          tabIndex={0}
        >
          test
          <svg
            className="places__sorting-arrow"
            height="4"
            width="7"
          >
            <use
              xlinkHref="#icon-arrow-select"
            />
          </svg>
        </span>
        <ul
          className="places__options places__options--custom places__options--opened"
          onClick={[Function]}
        >
          <li
            className="places__option"
            data-sort="Popular"
            tabIndex={0}
          >
            Popular
          </li>
          <li
            className="places__option"
            data-sort="Price: low to high"
            tabIndex={0}
          >
            Price: low to high
          </li>
          <li
            className="places__option"
            data-sort="Price: high to low"
            tabIndex={0}
          >
            Price: high to low
          </li>
          <li
            className="places__option"
            data-sort="Top rated first"
            tabIndex={0}
          >
            Top rated first
          </li>
        </ul>
      </form>
      <div
        className="cities__places-list places__list tabs__content"
      >
        <article
          className="cities__place-card place-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  200
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button place-card__bookmark-button--active"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "40%",
                    }
                  }
                >
                   
                </span>
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
              onClick={[Function]}
            >
              <a
                href="/detail.html"
              >
                Beautiful & luxurious apartment at great location
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              apartment
            </p>
          </div>
        </article>
        <article
          className="cities__place-card place-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  200
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button place-card__bookmark-button--active"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "40%",
                    }
                  }
                >
                   
                </span>
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
              onClick={[Function]}
            >
              <a
                href="/detail.html"
              >
                Beautiful & luxurious apartment at great location
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              apartment
            </p>
          </div>
        </article>
      </div>
    </section>
    <div
      className="cities__right-section"
    >
      <section
        className="cities__map map"
        id="map"
      />
    </div>
  </div>
</div>
`;
